test/stdlib/symbol-visibility.test-sh

```diff
@@ -0,0 +1,19 @@
+// Ensure that we do not export any weak symbols from the dylibs.
+//
+// Weak symbols require additional work from the loader to resolve the symbol at
+// load time and can cause ODR violations as well as unexpected symbol
+// satisfaction because the weak symbol may be used from a separate module.
+
+// RUN: %empty-directory(%t)
+
+// RUN: %llvm-nm --defined-only --extern-only --demangle %platform-dylib-dir/%target-library-name(swiftCore) > %t/all.txt
+// RUN: %llvm-nm --defined-only --extern-only --no-weak --demangle %platform-dylib-dir/%target-library-name(swiftCore) > %t/no-weak.txt
+// RUN: diff %t/all.txt %t/no-weak.txt
+
+// RUN: %llvm-nm --defined-only --extern-only --demangle %platform-dylib-dir/%target-library-name(swiftRemoteMirror) > %t/all.txt
+// RUN: %llvm-nm --defined-only --extern-only --no-weak --demangle %platform-dylib-dir/%target-library-name(swiftRemoteMirror) > %t/no-weak.txt
+// RUN: diff %t/all.txt %t/no-weak.txt
```

Comment:
It could be worth to use `diff -u` instead -- this way in the start of the output there are the names of the files being diffed, so it's easier to understand which command failed from the log